src/main/java/com/klodnicki/Bike/v2/service/RentBikeService.java

```diff
@@ -0,0 +1,163 @@
+
+        Rent rent = new Rent(LocalDateTime.now(), null, bike, user, null, daysOfRent);
+        rentRepository.save(rent);
+
+        return new RentResponseDTO(
+                rent.getId(),
+                rent.getRentalStartTime(),
+                null,
+                rent.getDaysOfRent(),
+                modelMapper.map(rent, BikeForNormalUserResponseDTO.class),
+                modelMapper.map(rent, UserForNormalUserResponseDTO.class), null);
+    }
+
+    @Override
+    public ChargingStation addBikeToList(Long chargingStationId, Bike bike) {
+        ChargingStation chargingStation = findStationById(chargingStationId);
+        chargingStation.getBikeList().add(bike);
+        //In JPA, only the owning side of the relationship is used when writing to the database.
+        //which means this will not be saved in database. Charging station still will have an empty bike list.
+        //because ChargingStation is NOT the owning-side. Therefore, I must add line below to fix this problem
+        bike.setChargingStation(chargingStation);
+
+        return chargingStationService.save(chargingStation);
+    }
+
+    @Override
+    @Transactional
+    public ResponseEntity<?> returnVehicle(Long rentId, Long returnChargingStationId, Long bikeId) {
+        Bike bike = bikeService.findBikeById(bikeId);
+        ChargingStation returnChargingStation = findStationById(returnChargingStationId);
+        Rent rent = findRentById(rentId);
+        User user = findUserById(bike.getUser().getId());
+
+        bike.setRented(false);
+        bike.setAmountToBePaid(countRentalCost(rentId));
+        bike.setUser(null);
+
+        returnChargingStation.getBikeList().add(bike); //Bike is an owning side, so I must add below line to save it in db
+        bike.setChargingStation(returnChargingStation);
+
+        returnChargingStation.setFreeSlots(returnChargingStation.getFreeSlots() - 1);
+
+        user.setBalance(user.getBalance() - countRentalCost(rentId)); //reducing the user balance by rental cost
+
+        rent.setRentalEndTime(LocalDateTime.now());
+        rent.setChargingStation(returnChargingStation);
+
+        bikeService.save(bike);
+        chargingStationService.save(returnChargingStation);
+        userService.save(user);
+        rentRepository.save(rent);
+
+        return new ResponseEntity<>("Bike successfully returned.", HttpStatus.OK);
+    }
+
+    private double countRentalCost(Long rentId) {
+        Rent rent = findRentById(rentId);
+        int rentalDays = rent.getDaysOfRent();
+        return rentalDays * 10;
+    }
+
+    private Rent findRentById(Long id) {
+        return rentRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+    }
+
+    private User findUserById(Long id) {
+        return userService.findUserById(id);
+    }
+
+    private ChargingStation findStationById(Long returnChargingStationId) {
+        return chargingStationService.findStationById(returnChargingStationId);
+    }
```

Comment:
Those methods doesn't do anything. 
Just use `userService.findUserById(id);` whenever you need it as you have access to this method.Same applies to the method below it.

Reply:
what do you mean they don't do anything?
they return me objects, I use those methods in several other methods in this class

Reply:
You can simply use i.e. `userService.findUserById(id);` instead of creating a private method which does it

Reply:
Ok! understood! :)